clients/src/main/java/org/apache/kafka/common/serialization/Serialization.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership. The ASF licenses this file
+ * to You under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.apache.kafka.common.serialization;
+
+/**
+ * The interface for wrapping a serializer and deserializer for the given data type.
+ *
+ * @param <T>
+ */
+public interface Serialization<T> {
+
+    Serializer<T> serializer();
```

Comment:
> The currently proposed interface has the advantage that you can use existing serializers and deserializers, but it has the downside you suggested.

Are the serializers and deserializers (like `LongSerializer`) used anywhere else other than Kafka Streams at the moment?  That is, would it be safe to merge them if we want to as part of this PR?

> [...] One question is: do we want `SerDe` to be in `o.a.kafka.common` even though it's not used anywhere but Kafka Streams?
Good point -- I'll leave the answer to @guozhangwang.  More interesting though is the previous point, i.e. if it is currently not used elsewhere, are we safe to collapse Sers and Desers (because it would have no collateral damage)?  IMHO the question of where the serdes reside (e.g. `o.a.kafka.common`) is a decoupled from the collapse yes/no question.  Location-wise I'm fine either way.